### PR TITLE
chore: add chromatic workflow

### LINKED ISSUE #21923: Add the ability to make screenshots tests for WDS components

### LINKED ISSUE #24525: [Task]: Integrate chromatic for storybook visual testing

### PR DESCRIPTION
**Pull request by jsartisan**
This PR adds the integration for chromatic which will help us catch visual bugs in the WDS components.
The way it works is, the PR won't allowed to merge till reviewers reviews all the changes in the components.
One thing to note is the workflow only works when there is a change in design system files.

Fixes #21923

### CODE REVIEW

**Review by sum35h — DISMISSED** [review]

File: .github/workflows/build-chromatic.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        with:
+          fetch-depth: 0
+          ref: refs/pull/${{ github.event.pull_request.number }}/merge
+
+      - name: Use Node.js 16.14.0
+        uses: actions/setup-node@v3
+        with:
+          node-version: "16.14.0"
+          cache: "yarn"
+          cache-dependency-path: "app/client/yarn.lock"
```
Comment (KelvinOm) [inline .github/workflows/build-chromatic.yml:24]: @sum35h @jsartisan Do we need save cache like in client build?

https://github.com/appsmithorg/appsmith/blob/release/.github/workflows/client-build.yml#L191-L197

File: .github/workflows/build-chromatic.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+      - name: Install Dependencies
+        working-directory: ./app/client/packages/storybook
+        run: yarn install --immutable
+        
+      - name: Publish to Chromatic
+        id: chromatic-publish
```
Comment (KelvinOm) [inline .github/workflows/build-chromatic.yml:31]: 1. `build-wds-preview.yml`
2. `wds-vercel-preview.yml`

If we started using chromatic, then we don't need these files anymore, right? Сan we use chromatic to publish a storybook after merging a PR?

File: app/client/packages/storybook/package.json [hunk 3]
```diff
@@ -6,7 +6,8 @@
   "license": "MIT",
   "scripts": {
     "storybook": "start-storybook -p 6006",
-    "build": "build-storybook"
+    "build": "build-storybook",
+    "chromatic": "chromatic --project-token CHROMATIC_PROJECT_TOKEN"
```
[the remark this answers is not in the document]
Comment (KelvinOm) [reply app/client/packages/storybook/package.json:10]: The question is not relevant.

**Comment by KelvinOm** [comment]
@jsartisan Can we somehow separate workspaces in chromatics for WDS and ADS? We are currently working in the appsmithorg/appsmith project, maybe we should create a separate project?

Now it is quite difficult to separate what relates to WDS and I can see a lot of builds that I do not know about.
Снимок экрана 2023-06-22 в 17 31 30

File: .github/workflows/build-chromatic.yml (hunk 1 above)
Comment (jsartisan) [reply .github/workflows/build-chromatic.yml:24]: I am also sure about if we need it. I added it because it was present in other workflow file as well.

Comment (KelvinOm) [reply .github/workflows/build-chromatic.yml:24]: @sum35h could you help us with this?

Comment (sum35h) [reply .github/workflows/build-chromatic.yml:24]: Lets skip the cache for now?

Comment (sum35h) [inline .github/workflows/build-chromatic.yml:24]:
```suggestion
```

Comment (sum35h) [reply .github/workflows/build-chromatic.yml]: Skipping saving build cache for now.

**Review by sum35h — DISMISSED** [review]

**Pull request merged**